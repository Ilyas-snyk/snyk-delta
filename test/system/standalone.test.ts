import { exec } from 'child_process';
import { sep } from 'path';
const stripAnsi = require('strip-ansi');
const main = './dist/index.js'.replace(/\//g, sep);

describe('`snyk-delta <...>`', () => {
  it('Shows error when missing mandatory parameters', (done) => {
    exec(
      `node ${main} --currentOrg uuid --currentProject uuid`,
      (err, stdout, stderr) => {
        expect(err).not.toBeNull();
        expect(stripAnsi(stderr)).toMatchSnapshot();
        expect(stripAnsi(stdout)).toMatchSnapshot();
      },
    ).on('exit', (code) => {
      expect(code).toEqual(2);
      done();
    });
  }, 30000);
});
