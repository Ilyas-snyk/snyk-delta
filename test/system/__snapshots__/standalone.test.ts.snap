// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-delta <...>\` Shows error when missing mandatory parameters 1`] = `
"BadInputError: In 'standalone' mode --currentProject, --currentOrg, --baselineOrg and --baselineProject are required.
Please review the available documentation via -h or the README
"
`;

exports[`\`snyk-delta <...>\` Shows error when missing mandatory parameters 2`] = `
"Hint: use debug mode -d for more information
"
`;
